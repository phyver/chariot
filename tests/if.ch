data
  ifexp('a) where At : 'a -> ifexp('a)
                | If : ifexp('a) ->  ifexp('a) -> ifexp('a) -> ifexp('a)

-- :set show_coherent_loops true
-- :set show_bad_loops true
-- :set show_all_steps true
-- :set incremental_SCP false

val nm : ifexp('a) -> ifexp('a)
  | nm (At a) = At a
  | nm (If (At a) y z) = If (At a) (nm y) (nm z)
  | nm (If (If u v w) y z) = nm (If u (nm (If v y z)) (nm (If w y z)))

-- :show functions


