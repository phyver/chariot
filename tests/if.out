>> option verbose is set to 1
>> inductive type ifexp was succesfully defined
>> Typing for nm successful
>> the definition for nm is complete
!! warning: the definition nm is NOT provably total (weight_bound: 2, depth_bound: 2)

